src/retrace/retrace.py

```diff
@@ -474,11 +519,10 @@ def is_package_known(package_nvr, arch, releaseid=None):
                 p = get_package_by_nevra(db, n, e or 0, v, r, a)
                 if p is not None:
                     return True
-        else:
-            # Try with noarch
-            p = get_package_by_nevra(db, n, e or 0, v, r, "noarch")
-            if p is not None:
-                return True
+                # Try with noarch
+                p = get_package_by_nevra(db, n, e or 0, v, r, "noarch")
+                if p is not None:
+                    return True
```

Comment:
This is a bit questionable(?) change. 

Before this was executed: 0 times or once.
After this is executed: 0 times, once or len(derived_archs) times

(Please correct me if I am wrong.)
https://docs.python.org/3.6/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops

Reply:
Ah, yes, silly me, thanks. Reverted.